src/main/java/org/jetbrains/research/refactorinsight/folding/JavaFileFinder.java

```diff
@@ -0,0 +1,81 @@
+package org.jetbrains.research.refactorinsight.folding;
+
+import com.intellij.psi.*;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+
+public class JavaFileFinder {
+  // Suppresses default constructor, ensuring non-instantiability.
+  private JavaFileFinder() {
+  }
+
+  @Nullable
+  public static PsiMethod findMethod(@NotNull PsiFile psiFile, @NotNull String qualifiedName) {
+    int classQualifiedNameEnd = qualifiedName.lastIndexOf('.');
+    int parametersListStart = qualifiedName.indexOf('(');
+    int parametersListEnd = qualifiedName.indexOf(')');
+    assert parametersListEnd + 1 == qualifiedName.length();
+
+    PsiClass psiClass = findClass(psiFile, qualifiedName.substring(0, classQualifiedNameEnd));
+    if (psiClass != null) {
+      PsiMethod[] psiMethods = psiClass.findMethodsByName(
+          qualifiedName.substring(classQualifiedNameEnd + 1, parametersListStart), false);
+      if (psiMethods.length > 0) {
+        String[] searchedMethodParameters =
+            parametersListStart + 1 < parametersListEnd
+                ? qualifiedName.substring(parametersListStart + 1, parametersListEnd).split(", ")
+                : new String[]{};
+        for (PsiMethod psiMethod : psiMethods) {
+          String[] methodParameters =
+              Arrays.stream(psiMethod.getParameterList().getParameters())
+                  .map(PsiParameter::getType)
+                  .map(PsiType::getPresentableText)
+                  .toArray(String[]::new);
+          if (Arrays.equals(methodParameters, searchedMethodParameters)) {
```

Comment:
Implement a method for signature calculation in `PsiUtils` and then just check methods' signatures here